process/peer/process.go

```diff
@@ -421,14 +419,8 @@ func (vs *validatorStatistics) getValidatorDataFromLeaves(
 	}
 	validators[core.MetachainShardId] = make([]*state.ValidatorInfo, 0)
 
-	sliceLeaves := vs.convertMapToSortedSlice(leaves)
-
-	sort.Slice(sliceLeaves, func(i, j int) bool {
-		return bytes.Compare(sliceLeaves[i], sliceLeaves[j]) < 0
-	})
-
-	for _, pa := range sliceLeaves {
-		peerAccount, err := vs.unmarshalPeer(pa)
+	for pa := range leavesChannel {
```

Comment:
Interesting. This channel should cause deterministic outputs on each elrond-go node since the trie nodes traversal is deterministic. The sort you remove should no longer be necessary. However, we might get into backwards incompatibility issues. We have to check.

Reply:
Will check for backwards compatibility issues.